resilience4j-bulkhead/src/main/java/io/github/resilience4j/bulkhead/Bulkhead.java

```diff
@@ -557,6 +557,17 @@ default void executeRunnable(Runnable runnable) {
         return decorateCompletionStage(this, supplier).get();
     }
 
+    /**
+     * Decorates and executes the decorated Future.
+     *
+     * @param supplier the original Future
+     * @param <T>      the type of results supplied by this supplier
+     * @return the decorated CompletionStage.
```

Comment:
I do not feel we should do that at all , it is either the wrapped future and let the user execute it or we should provide an optional executor to be used for that